src/main/kotlin/org/rust/ide/intentions/JoinParameterListIntention.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.intentions
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import org.rust.lang.core.psi.RsValueParameter
+import org.rust.lang.core.psi.RsValueParameterList
+import org.rust.lang.core.psi.ext.RsElement
+
+abstract class JoinListIntention<TList : RsElement, TElement : RsElement>(
+    listClass: Class<TList>,
+    elementClass: Class<TElement>,
+    intentionText: String
+) : ChopListIntention<TList, TElement>(listClass, elementClass, intentionText) {
+
+    override fun findApplicableContext(project: Project, editor: Editor, element: PsiElement): TList? {
+        val list = element.listContext ?: return null
+        val elements = list.elements
+        if (elements.isEmpty() || !hasLineBreakBefore(elements.first()) && elements.none { hasLineBreakAfter(it) }) {
+            return null
+        }
+        return list
+    }
+
+    override fun invoke(project: Project, editor: Editor, ctx: TList) {
+        val document = editor.document
+        val elements = ctx.elements
+
+        nextBreak(elements.last())?.let { document.deleteString(it.textRange.startOffset, it.textRange.endOffset) }
```

Comment:
Kotlin has extension properties `PsiElement#startOffset` and `PsiElement#endOffset`
Maybe we want to have them too? Looks like they are rather useful